src/libstore/parsed-derivations.cc

```diff
@@ -122,7 +122,7 @@ bool ParsedDerivation::willBuildLocally(Store & localStore) const
 
 bool ParsedDerivation::substitutesAllowed() const
 {
-    return getBoolAttr("allowSubstitutes", true);
+    return settings.alwaysAllowSubstitutes ? true : getBoolAttr("allowSubstitutes", true);
```

Comment:
This is really not a property of a "parsed derivation", but rather a property of the desired scheduling approach.
Not happy about that, but it's a structural problem, not a new thing.